ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,540 @@
+  SELECT create_distributed_table('impressions', 'company_id');
+
+The :ref:`create_distributed_table` function informs Citus that a table should be distributed and that future incoming queries to those tables should be planned for distributed execution. The function also creates shards for the table on worker nodes, allowing queries to be routed there.
+
+The next step is loading sample data into the cluster.
+
+.. code-block:: bash
+
+  # download and ingest datasets from the shell
+
+  for dataset in companies campaigns ads clicks impressions; do
+    curl -O https://examples.citusdata.com/tutorial/${dataset}.csv
+  done
+
+.. note::
+
+  **If you are using Docker,** you should use the :code:`docker cp` command to copy the files into the Docker container.
+
+  .. code-block:: bash
+
+    docker cp companies.csv citus_master:.
+    docker cp campaigns.csv citus_master:.
+    docker cp ads.csv citus_master:.
+
+Being an extension of PostgreSQL, Citus supports bulk loading with the COPY command. Use it to ingest the data you downloaded, and make sure that you specify the correct file path if you downloaded the file to some other location.
+
+.. code-block:: psql
+
+  \copy companies
+    from 'companies.csv' with csv;
+  \copy campaigns
+    from 'campaigns.csv' with csv;
+  \copy ads (id, company_id, campaign_id, name, image_url, target_url,
+             impressions_count, clicks_count, created_at, updated_at)
+    from 'ads.csv' with csv;
+  \copy clicks
+    from 'clicks.csv' with csv;
+  \copy impressions
+    from 'impressions.csv' with csv;
+
+Querying the Cluster
+--------------------
+
+Multi-tenant applications in Citus can run any SQL queries, as long as the queries include the tenant id as a filter condition. For instance, consider a simple query to find the top campaigns with highest budget for company five.
+
+::
+
+  -- Top ten campaigns with highest budget for a company
+
+  SELECT name, cost_model, state, monthly_budget
+    FROM campaigns
+   WHERE company_id = 5
+  ORDER BY monthly_budget DESC
+  LIMIT 10;
+
+Updates work too. Let's double the budget for all campaigns!
+
+::
+
+  UPDATE campaigns
+  SET monthly_budget = monthly_budget*2
+  WHERE company_id = 5;
+
+In all these queries, the filter routes SQL execution directly inside a worker. Full SQL support is available once queries are "pushed down" to a worker like this. For example, how about transactions in our distributed database? No problem:
+
+::
+
+  -- transactionally remove campaign 46 and all its ads
+
+  BEGIN;
+  DELETE from campaigns where id = 46 AND company_id = 5;
+  DELETE from ads where campaign_id = 46 AND company_id = 5;
+  COMMIT;
+
+When people scale applications with NoSQL databases they often miss the lack of transactions and joins. We already saw a join query when discussing distribution columns, but here's another to combine information from campaigns and ads.
+
+::
+
+  -- Total campaign budget vs expense this month
```

Comment:
Should we add a super complex query (with window functions / distinct / outer joins / where clause subqueries etc.) to inform the reader that any SQL will work?